src/app/test-record/test-record.mapper.ts

```diff
@@ -251,45 +145,60 @@ export class TestRecordMapper {
       testResultMapped.testExpiryDate = undefined;
     }
 
-    if (testTypesApplicable.defectsApplicable.hasOwnProperty(testTypeMapped.testTypeId)) {
-      testTypeMapped.defects = undefined;
-    }
-
-    if (!testTypesApplicable.testSectionApplicable1.hasOwnProperty(testTypeMapped.testTypeId)) {
+    if (!testTypesApplicable.anniversaryDateApplicable[testTypeMapped.testTypeId]) {
       testTypeMapped.certificateNumber = undefined;
       testTypeMapped.testExpiryDate = undefined;
       testTypeMapped.testAnniversaryDate = undefined;
     }
 
-    if (!testTypesApplicable.testSectionApplicable2.hasOwnProperty(testTypeMapped.testTypeId)) {
+    if (
```

Comment:
Same for this please. Could in the same ticket.

Reply:
https://jira.dvsacloud.uk/browse/CVSB-16228